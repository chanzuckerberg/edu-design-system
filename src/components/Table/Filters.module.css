/**
 * Filters container (fieldset)
 */
.filters-fieldset {
  margin: 0;
}

.filters-fieldset__legend {
  font: var(--eds-theme-typography-title-sm);
  margin-bottom: var(--eds-size-3);
}

.filters-fieldset__checkboxes > :not(:last-child) {
  margin-bottom: var(--eds-size-2-and-half);
}

/**
 * Filters popover overrides
 */
.filters-popover {
  padding: var(--eds-size-3);
}

.footer__button-group {
  margin-top: var(--eds-size-3);
  display: flex;
  justify-content: flex-end;
}

/**
 * Filters Button
 */
.filters-button {
  gap: var(--eds-size-1);
}
