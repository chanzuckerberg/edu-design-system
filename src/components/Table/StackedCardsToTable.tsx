import debounce from 'lodash/debounce';
import React, { useEffect, useState } from 'react';
import { Card, Table, Text } from '../../../src';

import breakpoint from '../../../src/design-tokens/tier-1-definitions/breakpoints';

/**
 * Demonstrates usage of stacked `<Card>` components for smaller breakpoints and `<Table>` component for larger breakpoints.
 */
export const StackedCardsToTable = () => {
  const [isTable, setIsTable] = useState(false);
  /**
   * Display data as cards if mobile, table if not.
   */
  const tableBreakpoint = parseInt(breakpoint['eds-bp-md'], 10) * 16;
  const updateScreenSize = debounce(
    () => {
      if (window.innerWidth >= tableBreakpoint && !isTable) {
        setIsTable(true);
      }
      if (window.innerWidth < tableBreakpoint && isTable) {
        setIsTable(false);
      }
    },
    200,
    { leading: true },
  );
  useEffect(() => {
    updateScreenSize();
    window.addEventListener('resize', updateScreenSize);
    return () => {
      window.removeEventListener('resize', updateScreenSize);
    };
  }, [updateScreenSize]);

  const checkpointProgresses = [
    {
      student: 'Araya, Raquel',
      status: 'Stop and Revise',
      cogSkill: 3,
      timeSubmitted: '2 hours ago',
    },
    {
      student: 'Jesse Banet',
      status: 'Stop and Revise',
      cogSkill: 1,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Julie Davis',
      status: 'Working',
      cogSkill: 0,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Hussain, Adnan',
      status: 'Review Feedback',
      cogSkill: 0,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Ilango, Megha',
      status: 'Needs Feedback',
      cogSkill: 3,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Jaffer, Arman',
      status: 'Needs Feedback',
      cogSkill: 1,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Kang, Michelle',
      status: 'Review Feedback',
      cogSkill: 1,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Lewine, Chris',
      status: 'Continue Working',
      cogSkill: 2,
      timeSubmitted: '3 days ago',
    },
    {
      student: 'Luo, Celia',
      status: 'Review Feedback',
      cogSkill: 1,
      timeSubmitted: '3 days ago',
    },
  ];

  return (
    <div>
      {isTable && (
        <Table>
          <Table.Header>
            <Table.Row variant="header">
              <Table.HeaderCell>Students</Table.HeaderCell>
              <Table.HeaderCell>Status</Table.HeaderCell>
              <Table.HeaderCell>Cog skill</Table.HeaderCell>
              <Table.HeaderCell>Time submitted</Table.HeaderCell>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            {checkpointProgresses.map((progress) => (
              <Table.Row key={`progress-${progress.student}`}>
                <Table.Cell>{progress.student}</Table.Cell>
                <Table.Cell>{progress.status}</Table.Cell>
                <Table.Cell>{progress.cogSkill}</Table.Cell>
                <Table.Cell>{progress.timeSubmitted}</Table.Cell>
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      )}
      {!isTable &&
        checkpointProgresses.map((progress) => (
          <Card className="mb-4 py-8" key={`progress-${progress.student}`}>
            <Card.Header>
              <Text as="p" preset="body-lg">
                {progress.student}
              </Text>
            </Card.Header>
            <Card.Body>
              <div className="flex justify-between">
                <Text as="p" preset="body-sm">
                  Status
                </Text>
                <Text as="p" preset="body-sm">
                  {progress.status}
                </Text>
              </div>
              <div className="flex justify-between">
                <Text as="p" preset="body-sm">
                  Cog skill
                </Text>
                <Text as="p" preset="body-sm">
                  {progress.cogSkill}
                </Text>
              </div>
              <div className="flex justify-between">
                <Text as="p" preset="body-sm">
                  Time submitted
                </Text>
                <Text as="p" preset="body-sm">
                  {progress.timeSubmitted}
                </Text>
              </div>
            </Card.Body>
          </Card>
        ))}
    </div>
  );
};
