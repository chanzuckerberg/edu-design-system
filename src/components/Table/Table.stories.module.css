.table__row--zebra:hover {
  background-color: var(--eds-theme-color-background-neutral-default-hover);
}
