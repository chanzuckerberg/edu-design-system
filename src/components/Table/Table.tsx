import clsx from 'clsx';
import type { MouseEventHandler, ReactNode } from 'react';
import React from 'react';
import Button from '../Button';
import Icon, { type IconName } from '../Icon';
import styles from './Table.module.css';

type TableBodyProps = {
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
};

type TableCaptionProps = React.HTMLAttributes<HTMLTableCaptionElement> & {
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
};

type TableCellProps = React.TdHTMLAttributes<HTMLTableCellElement> & {
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /*
   * This attribute contains a non-negative integer value that indicates for how many columns the cell extends
   */
  colSpan?: number;
  /**
   * This attribute contains a list of space-separated strings, each corresponding to the id attribute of the `th` elements that apply to this element.
   */
  headers?: string;
  /**
   * HTML id for the component
   */
  id?: string;
  /**
   * This attribute contains a non-negative integer value that indicates for how many rows the cell extends
   */
  rowSpan?: number;
  /**
   * This enumerated attribute defines the cells that the header (defined in the `th`) element relates to.
   */
  scope?: 'row' | 'col' | 'colgroup';
};

type TableFooterProps = {
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
};

type TableHeaderProps = React.HTMLAttributes<HTMLTableSectionElement> & {
  /**
   * Child node(s) that can be nested inside component
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
};
type TableHeaderCellProps = React.ThHTMLAttributes<HTMLTableCellElement> & {
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * This attribute contains a non-negative integer value that indicates for how many columns the cell extends
   */
  colSpan?: number;
  /**
   * This attribute contains a list of space-separated strings, each corresponding to the id attribute of the `th` elements that apply to this element.
   */
  headers?: string;
  /**
   * HTML id for the component
   */
  id?: string;
  /**
   * This attribute contains a non-negative integer value that indicates for how many rows the cell extends
   */
  rowSpan?: number;
  /**
   * Callback called when the sort button is clicked.
   */
  onSortClick?: MouseEventHandler;
  /**
   * This enumerated attribute defines the cells that the header (defined in the `th`) element relates to.
   */
  scope?: 'row' | 'col' | 'colgroup';
  /**
   * The direction the selected column will be sorted.
   */
  sortDirection?: SortDirectionsType;
  /**
   * Variant for table header cell within table body.
   * Matches <Table.Cell> padding for alignment.
   */
  variant?: 'body';
};

type TableRowProps = React.HTMLAttributes<HTMLTableRowElement> & {
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Header variant has a darker bottom border to distinguish itself as a header.
   */
  variant?: 'header';
};

type TableProps = React.TableHTMLAttributes<HTMLTableElement> & {
  /**
   * Child node(s) that can be nested inside component.
   */
  children: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
};

/**
 * HTML `tbody` of the `Table` component
 */
const TableBody = ({ children, ...other }: TableBodyProps) => {
  return <tbody {...other}>{children}</tbody>;
};

/**
 * HTML caption cell for the `Table` component.
 * Must be first descendant of the `Table` component.
 */
const TableCaption = ({ children, ...other }: TableCaptionProps) => {
  return <caption {...other}>{children}</caption>;
};

/**
 * HTML table cell of the `Table` component
 */
const TableCell = ({ children, className, ...other }: TableCellProps) => {
  const componentClassName = clsx(styles['table-cell'], className);

  return (
    <td className={componentClassName} {...other}>
      {children}
    </td>
  );
};

/**
 * HTML `tfoot` of the `Table` component
 */
const TableFooter = ({ children, className, ...other }: TableFooterProps) => {
  return (
    <tfoot className={className} {...other}>
      {children}
    </tfoot>
  );
};

/**
 * HTML `thead` of the `Table` component
 */
const TableHeader = ({ children, className, ...other }: TableHeaderProps) => {
  return (
    <thead className={className} {...other}>
      {children}
    </thead>
  );
};

const SORT_DIRECTIONS = ['ascending', 'descending', 'default'] as const;

export type SortDirectionsType = (typeof SORT_DIRECTIONS)[number];

/**
 * `import {TableHeaderCell} from "@chanzuckerberg/eds";`
 *
 * HTML `th` cell of the `Table` component
 */
const TableHeaderCell = ({
  children,
  className,
  onSortClick,
  sortDirection,
  variant,
  ...other
}: TableHeaderCellProps) => {
  const componentClassName = clsx(
    styles['table-header-cell'],
    variant === 'body' && styles['table-header-cell--body'],
    className,
  );

  const icon: IconName =
    sortDirection === 'ascending'
      ? 'arrow-narrow-up'
      : sortDirection === 'descending'
      ? 'arrow-narrow-down'
      : 'unfold-more';

  const iconTitle =
    sortDirection === 'ascending'
      ? 'Sorted, ascending'
      : sortDirection === 'descending'
      ? 'Sorted, descending'
      : 'Sort';

  return (
    <th
      aria-sort={
        sortDirection === 'ascending' || sortDirection === 'descending'
          ? sortDirection
          : undefined
      }
      className={componentClassName}
      {...other}
    >
      {sortDirection ? (
        <Button
          className={clsx(styles['table-header-cell__sort-button'])}
          onClick={onSortClick}
          variant="link"
        >
          {children}
          <Icon
            name={icon}
            purpose="informative"
            size="1rem"
            title={iconTitle}
          />
        </Button>
      ) : (
        children
      )}
    </th>
  );
};

/**
 * `import {TableRow} from "@chanzuckerberg/eds";`
 *
 * HTML `tr` of the `Table` component
 */
export const TableRow = ({
  children,
  className,
  variant,
  ...other
}: TableRowProps) => {
  const componentClassName = clsx(
    styles['table-row'],
    variant === 'header' && styles['table-row--header'],
    className,
  );

  return (
    <tr className={componentClassName} {...other}>
      {children}
    </tr>
  );
};

/**
 * `import {Table} from "@chanzuckerberg/eds";`
 *
 * HTML table component. Allows for the structure and layout of a standard HTML table, with styles to support theming.
 */
export const Table = ({ children, className, ...other }: TableProps) => {
  const componentClassName = clsx(styles['table'], className);
  return (
    <table className={componentClassName} {...other}>
      {children}
    </table>
  );
};

Table.displayName = 'Table';
TableBody.displayName = 'Table.Body';
TableCell.displayName = 'Table.Cell';
TableFooter.displayName = 'Table.Footer';
TableHeader.displayName = 'Table.Header';
TableHeaderCell.displayName = 'Table.HeaderCell';
TableRow.displayName = 'Table.Row';
TableCaption.displayName = 'Table.Caption';

Table.Body = TableBody;
Table.Cell = TableCell;
Table.Footer = TableFooter;
Table.Header = TableHeader;
Table.HeaderCell = TableHeaderCell;
Table.Row = TableRow;
Table.Caption = TableCaption;
