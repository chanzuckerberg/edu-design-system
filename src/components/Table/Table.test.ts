import { generateSnapshots } from '@chanzuckerberg/story-utils';
import * as stories from './Table.stories';
import type { StoryFile } from '../../../.storybook/utility-types';

describe('<Table />', () => {
  generateSnapshots(stories as StoryFile);
});
