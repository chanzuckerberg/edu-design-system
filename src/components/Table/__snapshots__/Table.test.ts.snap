// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> AlignTableCellContentCenter story renders snapshot 1`] = `
<table
  class="table"
>
  <caption
    class="table__caption u-is-vishidden"
  >
    This is a table caption and it is required
  </caption>
  <thead
    class="table__header"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__cell table__cell--align-center"
      >
        Name
      </th>
      <th
        class="table__cell table__cell--align-center"
      >
        Status
      </th>
      <th
        class="table__cell table__cell--align-center"
      >
        Chart
      </th>
      <th
        class="table__cell table__cell--align-center"
      >
        Window
      </th>
      <th
        class="table__cell table__cell--align-center"
      >
        Last Update
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-center"
        data-label="Name"
      >
        Table Row 1, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Status"
      >
        Table Row 1, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Chart"
      >
        Table Row 1, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Window"
      >
        Table Row 1, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Last Update"
      >
        Table Row 1, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-center"
        data-label="Name"
      >
        Table Row 2, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Status"
      >
        Table Row 2, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Chart"
      >
        Table Row 2, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Window"
      >
        Table Row 2, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Last Update"
      >
        Table Row 2, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-center"
        data-label="Name"
      >
        Table Row 3, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Status"
      >
        Table Row 3, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Chart"
      >
        Table Row 3, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Window"
      >
        Table Row 3, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Last Update"
      >
        Table Row 3, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-center"
        data-label="Name"
      >
        Table Row 4, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Status"
      >
        Table Row 4, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Chart"
      >
        Table Row 4, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Window"
      >
        Table Row 4, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-center"
        data-label="Last Update"
      >
        Table Row 4, Table Cell 5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> AlignTableCellContentRight story renders snapshot 1`] = `
<table
  class="table"
>
  <caption
    class="table__caption u-is-vishidden"
  >
    This is a table caption and it is required
  </caption>
  <thead
    class="table__header"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__cell table__cell--align-right"
      >
        Name
      </th>
      <th
        class="table__cell table__cell--align-right"
      >
        Status
      </th>
      <th
        class="table__cell table__cell--align-right"
      >
        Chart
      </th>
      <th
        class="table__cell table__cell--align-right"
      >
        Window
      </th>
      <th
        class="table__cell table__cell--align-right"
      >
        Last Update
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-right"
        data-label="Name"
      >
        Table Row 1, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Status"
      >
        Table Row 1, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Chart"
      >
        Table Row 1, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Window"
      >
        Table Row 1, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Last Update"
      >
        Table Row 1, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-right"
        data-label="Name"
      >
        Table Row 2, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Status"
      >
        Table Row 2, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Chart"
      >
        Table Row 2, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Window"
      >
        Table Row 2, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Last Update"
      >
        Table Row 2, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-right"
        data-label="Name"
      >
        Table Row 3, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Status"
      >
        Table Row 3, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Chart"
      >
        Table Row 3, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Window"
      >
        Table Row 3, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Last Update"
      >
        Table Row 3, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell table__cell--align-right"
        data-label="Name"
      >
        Table Row 4, Table Cell 1
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Status"
      >
        Table Row 4, Table Cell 2
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Chart"
      >
        Table Row 4, Table Cell 3
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Window"
      >
        Table Row 4, Table Cell 4
      </td>
      <td
        class="table__cell table__cell--align-right"
        data-label="Last Update"
      >
        Table Row 4, Table Cell 5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> Default story renders snapshot 1`] = `
<table
  class="table"
>
  <caption
    class="table__caption u-is-vishidden"
  />
  <thead
    class="table__header"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__cell"
      >
        Name
      </th>
      <th
        class="table__cell"
      >
        Status
      </th>
      <th
        class="table__cell"
      >
        Chart
      </th>
      <th
        class="table__cell"
      >
        Window
      </th>
      <th
        class="table__cell"
      >
        Last Update
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> SortableInteractive story renders snapshot 1`] = `
<table
  class="table"
>
  <caption
    class="table__caption u-is-vishidden"
  >
    This is a table caption and it is required
  </caption>
  <thead
    class="table__header"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__header-cell"
      >
        <button
          class="table__header-cell-button"
        >
          Sortable
          <div
            class="table__header-cell-sort"
          >
            <svg
              class="icon table__header-cell-icon"
              fill="currentColor"
              height="1rem"
              role="img"
              style="--icon-size: 1rem;"
              width="1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="1"
              >
                Sort
              </title>
              <use
                xlink:href="test-file-stub#unfold-more"
              />
            </svg>
          </div>
        </button>
      </th>
      <th
        class="table__header-cell"
      >
        <div
          class="table__header-cell-contents"
        >
          Not sortable
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Value 1
      </td>
      <td
        class="table__cell"
      >
        Value A
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Value 3
      </td>
      <td
        class="table__cell"
      >
        Value B
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Value 2
      </td>
      <td
        class="table__cell"
      >
        Value C
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Value 4
      </td>
      <td
        class="table__cell"
      >
        Value D
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> Stacked story renders snapshot 1`] = `
<table
  class="table table--stacked"
>
  <caption
    class="table__caption u-is-vishidden"
  />
  <thead
    class="table__header"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__cell"
      >
        Name
      </th>
      <th
        class="table__cell"
      >
        Status
      </th>
      <th
        class="table__cell"
      >
        Chart
      </th>
      <th
        class="table__cell"
      >
        Window
      </th>
      <th
        class="table__cell"
      >
        Last Update
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
        data-heading="Name"
      >
        Table Row 1, Table Cell 1
      </td>
      <td
        class="table__cell"
        data-heading="Status"
      >
        Table Row 1, Table Cell 2
      </td>
      <td
        class="table__cell"
        data-heading="Chart"
      >
        Table Row 1, Table Cell 3
      </td>
      <td
        class="table__cell"
        data-heading="Window"
      >
        Table Row 1, Table Cell 4
      </td>
      <td
        class="table__cell"
        data-heading="Last Update"
      >
        Table Row 1, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
        data-heading="Name"
      >
        Table Row 2, Table Cell 1
      </td>
      <td
        class="table__cell"
        data-heading="Status"
      >
        Table Row 2, Table Cell 2
      </td>
      <td
        class="table__cell"
        data-heading="Chart"
      >
        Table Row 2, Table Cell 3
      </td>
      <td
        class="table__cell"
        data-heading="Window"
      >
        Table Row 2, Table Cell 4
      </td>
      <td
        class="table__cell"
        data-heading="Last Update"
      >
        Table Row 2, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
        data-heading="Name"
      >
        Table Row 3, Table Cell 1
      </td>
      <td
        class="table__cell"
        data-heading="Status"
      >
        Table Row 3, Table Cell 2
      </td>
      <td
        class="table__cell"
        data-heading="Chart"
      >
        Table Row 3, Table Cell 3
      </td>
      <td
        class="table__cell"
        data-heading="Window"
      >
        Table Row 3, Table Cell 4
      </td>
      <td
        class="table__cell"
        data-heading="Last Update"
      >
        Table Row 3, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
        data-heading="Name"
      >
        Table Row 4, Table Cell 1
      </td>
      <td
        class="table__cell"
        data-heading="Status"
      >
        Table Row 4, Table Cell 2
      </td>
      <td
        class="table__cell"
        data-heading="Chart"
      >
        Table Row 4, Table Cell 3
      </td>
      <td
        class="table__cell"
        data-heading="Window"
      >
        Table Row 4, Table Cell 4
      </td>
      <td
        class="table__cell"
        data-heading="Last Update"
      >
        Table Row 4, Table Cell 5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> Zebra story renders snapshot 1`] = `
<table
  class="table table--zebra"
>
  <caption
    class="table__caption u-is-vishidden"
  />
  <thead
    class="table__header"
  >
    <tr
      class="table__row"
    >
      <th
        class="table__cell"
      >
        Name
      </th>
      <th
        class="table__cell"
      >
        Status
      </th>
      <th
        class="table__cell"
      >
        Chart
      </th>
      <th
        class="table__cell"
      >
        Window
      </th>
      <th
        class="table__cell"
      >
        Last Update
      </th>
    </tr>
  </thead>
  <tbody
    class="table__body"
  >
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 1, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 2, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 3, Table Cell 5
      </td>
    </tr>
    <tr
      class="table__row"
    >
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 1
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 2
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 3
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 4
      </td>
      <td
        class="table__cell"
      >
        Table Row 4, Table Cell 5
      </td>
    </tr>
  </tbody>
</table>
`;
